### DISCUSSION TITLE
Doc for `HTTPTransport`

### DISCUSSION THREAD

**Discussion by stevenlis** [encode/httpx#2758] [Q&A]
I couldn't find any docs for `HTTPTransport` on www.python-httpx.org. Its docstring does not have any explanations for each param either.

Am I missing something?

I am trying to convert some of my requests code like the following and had to add a retry as I'm getting some 502 from time to time.
```python
s = requests.Session()
retries = Retry(total=5, backoff_factor=1, status_forcelist=[502])
s.mount('https://', HTTPAdapter(max_retries=retries))
r = s.get(url, params=params)
```

**Comment by lovelydinosaur** [comment]
This is the relevant part of our documentation...

https://www.python-httpx.org/advanced/#custom-transports

**Comment by stevenlis** [comment]
I already read it before posting. There are like only two lines of code in this doc, and I tried the following, but did not work.

```python
transport = httpx.HTTPTransport(retries=5)
client = httpx.Client(transport=transport)
r = client.get(url, params=params)
```
Is there anyway I could put a `backoff_factor ` and `status_forcelist`? I image a `status_forcelist` could be handled with a `httpx.HTTPStatusError`. But there is no doc for how to use it here.

**Comment by rvernica** [comment]
I found this page with examples on how to use HTTPTransport https://www.python-httpx.org/advanced/transports/ but a more detailed page showing all the arguments might help.
